### PR TITLE
Convert project to Sphinx cargo tool.

### PR DESCRIPTION
**Pull request by daithiocrualaoich**
Needs #16.

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -83,3 +83,16 @@ name = "vec_map"
 version = "0.6.0"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 
+[metadata]
```
Comment (daithiocrualaoich) [inline Cargo.lock]: Remove.

File: src/cmd.rs [hunk 2]
```diff
@@ -1,26 +1,24 @@
 use std::process::Command;
```
Comment (daithiocrualaoich) [inline src/cmd.rs]: We don't need the `call` variants without paths any more since cargo is not required.

File: README.md [hunk 3]
```diff
@@ -1,75 +1,100 @@
...
 
 
-Cargo GitHub Pages
-==================
-Forked from the [cargo-release] project by [Ning Sun]. This is the GitHub Pages
-functionality without the full release management.
+Cargo Sphinx
+============
+Cargo subcommand for generating and publishing Sphinx documentation to GitHub
+Pages.
```
Comment (passy) [inline README.md]: Could use a link to https://pages.github.com/

File: src/main.rs [hunk 4]
```diff
@@ -5,93 +5,106 @@ extern crate clap;
...
-        }
-    }
 
     // Generate and upload documentation.
-    println!("Building and exporting docs.");
-    try!(cargo::doc(dry_run));
+    println!("Building Sphinx docs.");
+    try!(call(vec!["make", "clean", "html"], docs_path, dry_run));
+
+    if push {
```
Comment (passy) [inline src/main.rs]: Bit too long if branch for my taste. What about a separate function just for the pushing?

File: src/main.rs [hunk 5]
```diff
@@ -5,93 +5,106 @@ extern crate clap;
...
-            .arg_from_usage("--doc-commit-message=[doc-branch] 'Git commit message to use'")
-            .arg_from_usage("--push-remote=[push-remote] 'Git remote for push'")
-            .arg_from_usage("--doc-branch=[doc-branch] 'Git branch to push documentation on'"))
+            .arg_from_usage("-p, --push 'Push generated documentation to git remote'")
+            .arg_from_usage("-s, --sign 'Sign git commit'")
+            .arg_from_usage("--docs-path=[docs-path] 'Path of Sphinx documentation to build'")
+            .arg_from_usage("--commit-message=[commit-message] 'Commit message for \
+                             documentation change'")
+            .arg_from_usage("--push-remote=[push-remote] 'Git remote to push'")
+            .arg_from_usage("--push-branch=[push-branch] 'Git branch to push documentation on'"))
```
Comment (passy) [inline src/main.rs]: Can we specify default values here that are visible on `--help`?

**Comment by passy** [comment]
:shipit: obviously.

File: src/main.rs (hunk 4 above)
Comment (daithiocrualaoich) [inline src/main.rs]: Oh yeah, this whole function is way long as well. I'll fix this up.

**Pull request merged**